File: gt4py-cffi/main.py
import numpy as np

import gt4py
from geometry import Geometry
from initialize import get_inout_fields
from stencil import double_laplacian, euler_step, time_swap
from tools.plotting import plot_two_ij_slices
from mainloop import march_in_time

if __name__ == '__main__':

    halo_size = 3
    geometry = Geometry(
        nx = 8, ny = 8, nz = 1,
        nhalo = halo_size,
        origin = (halo_size, halo_size, 0)
    )
    backend = 'gtx86'
    
    in_field, out_field = get_inout_fields(backend, geometry)
    
    nx, ny, nz, _, (ox, oy, oz) = geometry
    out_field = march_in_time(nx, ny, nz, ox, oy, oz, in_field, out_field)

File: gt4py-cffi/main.f90
program main

  use iso_fortran_env, only: real64
  use Geometry, only: Geometry_T
  use Initialize, only: get_inout_fields
  use PythonFortranInterface, only: march_in_time_interface

  implicit none

  integer, parameter :: halo_size = 3
  type(Geometry_T) :: geometry
  real(real64), allocatable :: in_field(:,:,:)
  real(real64), allocatable :: out_field(:,:,:)
  integer :: dim(3)
  integer :: i, j

  geometry = Geometry_T( &
       nx = 8, ny = 8, nz = 1, &
       nhalo = halo_size, &
       origin = [halo_size, halo_size, 0] &
       )

  call get_inout_fields(geometry, in_field, out_field, dim)
  call march_in_time_interface( &
       geometry%nx, geometry%ny, geometry%nz, &
       geometry%origin, &
       in_field, out_field, &
       dim & ! dimensions of in/out fields
       )
  do j = 1, dim(2)
     write(*,'(2x, (*(f4.2, 1x)))') (out_field(1,j,i), i = 1 , dim(3))
  end do

end program main
